gradoop-temporal/src/main/java/org/gradoop/temporal/model/impl/operators/metric/functions/BuildTemporalDegreeTree.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright © 2014 - 2021 Leipzig University (Database Research Group)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradoop.temporal.model.impl.operators.metric.functions;
+
+import org.apache.flink.api.common.functions.GroupReduceFunction;
+import org.apache.flink.api.common.functions.ReduceFunction;
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.api.java.tuple.Tuple3;
+import org.apache.flink.api.java.tuple.Tuple4;
+import org.apache.flink.util.Collector;
+import org.gradoop.common.model.impl.id.GradoopId;
+
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * A group reduce and reduce function collecting all edge time intervals of a vertex id to build a tree data
+ * structure.
+ */
+public class BuildTemporalDegreeTree
+  implements GroupReduceFunction<Tuple3<GradoopId, Long, Long>, Tuple2<GradoopId, TreeMap<Long, Integer>>>,
```

Comment:
implements on former line

Reply:
huh, sure ? I like leaving implements and the class together. We did this also before in other classes e.g. Overlaps.java